main.cpp

```diff
@@ -117,17 +151,17 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T ten(10 , "ten");                                             //6
+    T eleven(11, "eleven");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    ValueComparator f;                                            //7
+    auto* smaller = f.compare(&ten, &eleven);                              //8
+    std::cout << "the smaller one is << " << (smaller == nullptr ? "nullptr" : smaller->name) << std::endl; //9
```

Comment:
```
 9) correct the cout that uses smaller's member variable with some safe pointer usage.
    the possible return values of the compare() function should give you a hint.
    Also: if smaller == nullptr, make the cout statement explain the reasons that `f.compare()` might return nullptr.  
    hint: There are multiple reasons
```
- [ ] revise

Reply:
Converted to unique_ptr handling for equal values passed to the compare but it would be better if compare function returned the unique_ptr it seems in terms of ownership